src/Files.App/Shell/ShellLibraryEx.cs

```diff
@@ -0,0 +1,303 @@
+		/// <param name="kf">The known folder identifier.</param>
+		/// <param name="overwrite">if set to <c>true</c> [overwrite].</param>
+		public ShellLibraryEx(string libraryName, Shell32.KNOWNFOLDERID kf = Shell32.KNOWNFOLDERID.FOLDERID_Libraries, bool overwrite = false)
+		{
+			_lib = new Shell32.IShellLibrary();
+			_name = libraryName;
+			var item = _lib.SaveInKnownFolder(kf.Guid(), libraryName, overwrite ? Shell32.LIBRARYSAVEFLAGS.LSF_OVERRIDEEXISTING : Shell32.LIBRARYSAVEFLAGS.LSF_FAILIFTHERE);
+
+			Init(item);
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="ShellLibrary"/>Ex class.
+		/// </summary>
+		/// <param name="libraryName">Name of the library.</param>
+		/// <param name="parent">The parent.</param>
+		/// <param name="overwrite">if set to <c>true</c> [overwrite].</param>
+		public ShellLibraryEx(string libraryName, ShellFolder parent, bool overwrite = false)
+		{
+			_lib = new Shell32.IShellLibrary();
+			_name = libraryName;
+			var item = _lib.Save(parent.IShellItem, libraryName, overwrite ? Shell32.LIBRARYSAVEFLAGS.LSF_OVERRIDEEXISTING : Shell32.LIBRARYSAVEFLAGS.LSF_FAILIFTHERE);
+
+			Init(item);
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="ShellLibrary"/> class.
+		/// </summary>
+		/// <param name="iItem">The i item.</param>
+		/// <param name="readOnly">if set to <c>true</c> [read only].</param>
+		public ShellLibraryEx(Shell32.IShellItem iItem, bool readOnly = false)
+		{
+			_lib = new Shell32.IShellLibrary();
+			_lib.LoadLibraryFromItem(iItem, readOnly ? STGM.STGM_READ : STGM.STGM_READWRITE);
+
+			Init(iItem);
+		}
+
+		/// <summary>
+		/// Gets or sets the default target folder the library uses for save operations.
+		/// </summary>
+		/// <value>The default save folder.</value>
+		public ShellItem DefaultSaveFolder
+		{
+			get => Open(_lib.GetDefaultSaveFolder<Shell32.IShellItem>(Shell32.DEFAULTSAVEFOLDERTYPE.DSFT_DETECT));
+			set => _lib.SetDefaultSaveFolder(Shell32.DEFAULTSAVEFOLDERTYPE.DSFT_DETECT, value.IShellItem);
+		}
+
+		/// <summary>Gets the set of child folders that are contained in the library.</summary>
+		/// <value>A <see cref="ShellItemArray"/> containing the child folders.</value>
+		public ShellLibraryFolders Folders
+			=> _folders ??= GetFilteredFolders();
+
+		/// <summary>
+		/// Gets or sets a string that describes the location of the default icon.
+		/// The string must be formatted as
+		/// <c>ModuleFileName,ResourceIndex or ModuleFileName,-ResourceID</c>.
+		/// </summary>
+		/// <value>
+		/// The default icon location.
+		/// </value>
+		public IconLocation IconLocation
+		{
+			get
+			{
+				_ = IconLocation.TryParse(_lib.GetIcon(), out var l);
+				return l;
+			}
+			set => _lib.SetIcon(value.ToString());
+		}
+
+		/// <summary>
+		/// Gets the name relative to the parent for the item.
+		/// </summary>
+		public override string Name { get => _name; protected set => _name = value; }
+
```

Comment:
Shouldn't you split this into multiple lines?

Reply:
Fixed